index.bs

```diff
@@ -10791,6 +10578,71 @@ property-installation style as namespaces.)
     1.  Return |F|.
 </div>
 
+
+<h5 id="es-default-operations">Default operations</h5>
+
+Only [=regular operations=] which have a <dfn>corresponding default operation</dfn> defined below
+may be declared with a [{{Default}}] [=extended attribute=].
+
+
+<h6 id="es-default-tojson">Default toJSON operation</h5>
+
+The [=corresponding default operation=] of the “toJSON” operation is the [=default toJSON operation=].
+
+The [=return type=] of the [=default toJSON operation=] must be {{object}}.
+
+<div algorithm>
+
+    To invoke the <dfn export>default toJSON operation</dfn> of [=interface=] |I|,
+    run the the following steps:
+
+    1.  Let |map| be a new [=ordered map=].
+    1.  Let |stack| be the result of [=create an inheritance stack|creating an inheritance stack=]
+        for [=interface=] |I|.
+    1.  Invoke [=collect attribute values=], passing |stack| and |map|.
+    1.  Let |result| be [=!=] [=ObjectCreate=]([=%ObjectPrototype%=]).
+    1.  [=map/For each=] |key| → |value| of |map|,
+        1.  Let |k| be |key| [=converted to an ECMAScript value=].
+        1.  Let |v| be |value| [=converted to an ECMAScript value=].
+        1.  Perform [=!=] [=CreateDataProperty=](|result|, |k|, |v|).
+    1.  Return |result|.
+</div>
+
+<div algorithm>
+
+    To invoke the <dfn>collect attribute values</dfn> abstract operation
+    with [=stack=] |stack| and [=ordered map=] |map| as arguments,
+    run the the following steps:
+
+    1.  Let |I| be the result of invoking [=stack/pop=] from |stack|.
+    1.  [=list|For each=] [=implements statements=] where the left-hand side [=identifier=] references |I|:
+        1.  Let |rhs| be the [=interface=] represented by the right-hand side [=identifier=].
+        1.  Let |interfaces| be the result of [=create an inheritance stack|creating an inheritance stack=]
+            for [=interface=] |rhs|.
+        1.  invoke [=collect attribute values=], passing it |interfaces| and |map|.
+    1.  If a “toJSON” operation with a [{{Default}}] [=extended attribute=] is declared on |I|, then
+        [=list|for each=] [=exposed=] [=regular attribute=] |attr|
+        that is an [=interface member=] of |I|, in order:
+        1.  Let |id| be the [=identifier=] of |attr|.
+        1.  Let |value| be the underlying value of |attr|.
```

Comment:
Would adding something like the below in [§2.2.2 Attributes](https://heycam.github.io/webidl/#idl-attributes) do?

```
    To get the value of an [=attribute=] |attribute| declared in interface |I|,
    implemented by object |O|, run the following steps:

    1.  Let |R| be the result of performing the actions listed in the
        description of |attribute| that occur on getting
        (or those listed in the description of the inherited attribute,
        if this attribute is declared to [=inherit its getter=]),
        on |O| if |O| is not <emu-val>null</emu-val>. 
    1.  Return |R|.
```

How would you handle the "on |O| if |O| is not <emu-val>null</emu-val>" part keeping it in WebIDL, though?